api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -439,6 +441,27 @@ func validateLoadBalancerProfile(loadBalancerProfile *LoadBalancerProfile, fldPa
 	return allErrs
 }
 
+func validateAMCPVirtualNetwork(amcpSpec AzureManagedControlPlaneSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if amcpSpec.ResourceGroupName != amcpSpec.VirtualNetwork.ResourceGroup {
```

Comment:
Why are we only validating these vnet parameters if the resource group is different? It seems like we should validate these regardless.

And for context, this validation isn't strictly related to the bug, right?

Reply:
> Why are we only validating these vnet parameters if the resource group is different? 

Good thought. Now that I think of it, we should be validating regardless.

> this validation isn't strictly related to the bug

Correct. This validation originated out of my own testing.

Reply:
Added via 6b1927121631e5ca59ef6dd42e4005af4ad506e1